python/mspasspy/db/database.py

```diff
@@ -570,6 +571,150 @@ def update_ensemble_metadata(self, ensemble_object, include_undefined=False, exc
     # TODO: the following is not used when data is read from the database. We need
     #       link these metadata keys with the actual member variables just like the
     #       npts or t0 in TimeSeries.
+    def _load_collection_metadata(self,mspass_object, exclude_keys, include_undefined=False, collection=None):
+        """
+        Master Private Method
+
+        Reads metadata from a requested collection and loads standard attributes from collection to the data passed as mspass_object.
+        The method will only work if mspass_object has the collection_id attribute set to link it to a unique document in source.  
+
+        :param mspass_object:   data where the metadata is to be loaded
+        :type mspass_object:  must be TimeSeries, Seismogram, TimeSeriesEnsemble, or SeismogramEnsemble.
+        
+        :param exclude_keys: list of attributes that should not normally be loaded.
+        Default attributes not normally need that are loaded from stationxml.  Ignored if include_undefined is set True.
+        
+        :param include_undefined:  when true all data in the matching document are loaded.
+
+        :param collection: requested collection metadata should be loaded
+        :type collection: str 
+
+        :exception:  any detected errors will cause a MsPASSError to be thrown
+        (colleagues:  this may be wrong sphynx syntax for defining an exception)
+        """
+        if collection not in ["source", "site", "channel"]:
+            raise MsPASSError("collection param should be one of source, site or channel", ErrorSeverity.Invalid)
+
+        if not mspass_object.live:
+            raise MsPASSError("only live mspass object can load metadata", ErrorSeverity.Invalid)
+
+        if not isinstance(mspass_object, (TimeSeries, Seismogram)):
+            raise MsPASSError("only TimeSeries and Seismogram are supported", ErrorSeverity.Invalid)
+
+        if collection == 'channel' and isinstance(mspass_object, (Seismogram, SeismogramEnsemble)):
+            raise MsPASSError("channel data can not be loaded into Seismogram", ErrorSeverity.Invalid)
+        
+        # 1. get the metadata schema based on the mspass object type
+        wf_collection = 'wf_TimeSeries' if isinstance(mspass_object, TimeSeries) else 'wf_Seismogram'
+        object_type = self.database_schema[wf_collection].data_type()
```

Comment:
Object type is already determined by the input argument `mspass_object`. Why getting it from the database schema here? Also, the above line should be removed:
```wf_collection = 'wf_TimeSeries' if isinstance(mspass_object, TimeSeries) else 'wf_Seismogram'```